handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1062,7 +1062,15 @@ private SSLEngineResult wrap(ByteBufAllocator alloc, SSLEngine engine, ByteBuf i
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         boolean handshakeFailed = handshakePromise.cause() != null;
 
+        // Channel closed, we will generate 'ClosedChannelException' now.
         ClosedChannelException exception = new ClosedChannelException();
+
+        // Add a supressed exception if the handshake was not completed yet.
+        if (!isStateSet(STATE_HANDSHAKE_STARTED) || handshakePromise.isDone()) {
```

Comment:
In the previous round of feedback the logic was opposite. The above condition was for `ClosedChannelException`. Since now we changed the logic, we have to flip this condition as well. Consider: `isStateSet(STATE_HANDSHAKE_STARTED) && !handshakePromise.isDone()`.

Reply:
Yeah, sorry. My bad. ;/